techniques/system/systemUpdateCampaign/1.0/systemUpdateCampaign.st

```diff
@@ -0,0 +1,66 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2022 Normation SAS
+
+bundle agent rudder_system_update_campaign_&RudderUniqueID& {
+  vars:
+      "enabled"       string => "&ENABLE_UPDATE&";
+      "campaign_name" string => "&CAMPAIGN_NAME&";
+      "reboot"        string => "&REBOOT&";
+      "schedule"      string => "&SCHEDULE&";
+      "hours_splay"   string => "&HOURS_SPLAY&";
+      "campaign_id"   string => canonify("${campaign_name}");
+      "status_prefix" string => "rudder_system_update_campaign";
+      "reboot_opt"    string => "";
+
+    pass1::
+      "reboot_opt"    string => "--reboot",
+                          if => "reboot";
+      "cmd"           string => "--action=update --package_manager=${package_module_knowledge.platform_default} ${reboot_opt} --campaign_id=${campaign_id}",
+                          if => "update";
+      # If not update, report
+      # Will do nothing if report was already sent
+      "cmd"           string => "--action=report --campaign_id=${campaign_id}",
+                      unless => "update";
+
+    pass2::
+      # We can't pass JSON content as module output as if can exceed the size limit
+      # So we read it from file instead.
+      "report"        string => readfile("${rudder_system_update_module.system_update_report}", "0"),
+                          if => "${status_prefix}_report";
+
+  classes:
+      "enabled"   expression => strcmp("${enabled}", "enabled");
+      "reboot"    expression => strcmp("${reboot}", "enabled");
+      "now"       expression => strcmp("${hours_splay}", "0");
+      "update"    expression => "enabled.${schedule}.(now|schedule_simple_${campaign_id}_repaired)";
+
+      "pass3"     expression => "pass2";
+      "pass2"     expression => "pass1";
+      "pass1"     expression => "any";
+
+    pass2::
+      "call"      expression => usemodule("system_update.py", "${cmd}");
```

Comment:
we should have a protection here if the user manually calls the agent in the same timeperiod, so that we don't do 2 module calls at the same time or successively, like a class persisting for AGENT_RUN_INTERVAL preventing to usemodule

Reply:
we have a locking mechanism in the python module to address this already